File: app/schemas/activity.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ActivityCreate(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    type: str = Field(min_length=1, max_length=128)
    estimated_time: datetime
    location: str = Field(min_length=1, max_length=256)
    sponsor: str = Field(min_length=1, max_length=255)
    sponsor_contact: str = Field(min_length=1, max_length=128)
    sponsor_phone: str = Field(min_length=1, max_length=64)
    deadline: datetime
    designer_id: UUID | None = None


class ActivityResponse(BaseModel):
    id: UUID
    name: str
    type: str
    estimated_time: datetime
    location: str
    sponsor: str
    sponsor_contact: str | None = None
    sponsor_phone: str | None = None
    deadline: datetime
    status: str
    owner_id: UUID
    designer_id: UUID | None = None
    designer_name: str | None = None
    designer_phone: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ActivityPaginatedResponse(BaseModel):
    items: list[ActivityResponse]
    total: int


class ActivityListParams(BaseModel):
    status: str | None = None
    keyword: str | None = None
    date_from: datetime | None = None
    date_to: datetime | None = None
    page: int = Field(default=1, ge=1)
    size: int = Field(default=20, ge=1, le=100)


class StatusLogEntry(BaseModel):
    id: UUID
    from_status: str | None
    to_status: str
    operator_id: UUID
    comment: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/activity.ts
export interface ActivityCreate {
  name: string;
  type: string;
  estimated_time: string;
  location: string;
  sponsor: string;
  sponsor_contact: string;
  sponsor_phone: string;
  deadline: string;
  designer_id?: string;
}

export interface ActivityResponse {
  id: string;
  name: string;
  type: string;
  estimated_time: string;
  location: string;
  sponsor: string;
  sponsor_contact?: string | null;
  sponsor_phone?: string | null;
  deadline: string;
  status: string;
  owner_id: string;
  designer_id?: string | null;
  designer_name?: string | null;
  designer_phone?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ActivityPaginatedResponse {
  items: ActivityResponse[];
  total: number;
}

export interface ActivityListParams {
  status?: string;
  keyword?: string;
  date_from?: string;
  date_to?: string;
  sort?: string;
  page?: number;
  size?: number;
  tab?: string;
}

export interface StatusLogEntry {
  id: string;
  from_status: string | null;
  to_status: string;
  operator_id: string;
  operator_name?: string | null;
  comment: string | null;
  created_at: string;
}
